ckanext/stadtzhharvest/harvester.py

```diff
@@ -1019,17 +1034,38 @@ def _create_notifications_for_deleted_datasets(self):
                 )
                 log.debug('Removing metadata dir `%s`' % metadata_dir)
                 shutil.rmtree(metadata_dir)
-                # only send notification if there is a package in CKAN
-                if model.Package.get(package_id):
-                    path = self._diff_path(package_id)
-                    with open(path, 'w') as deleted_info:
-                        deleted_info.write(
-                            "<!DOCTYPE html>\n<html>\n<body>\n"
-                            "<h2>Dataset deleted: <a href=\"%s/dataset/%s\">%s"
-                            "</a></h2></body></html>\n"
-                            % (self.CKAN_SITE_URL, package_id, package_id)
-                        )
-                    log.debug('Wrote deleted notification to file `%s`' % path)
+
+                # only continue if package exists in CKAN
+                dataset_name = (
+                    '%s%s' % (self.config['dataset_prefix'], package_id)
+                )
+                if not model.Package.get(dataset_name):
+                    log.debug('Package `%s` not found in CKAN' % dataset_name)
+                    continue
+
+                self._create_notification_for_deleted_dataset(package_id)
+                if self.config['delete_missing_datasets']:
```

Comment:
Please check the value of `delete_missing_datasets` in `validate_config()` and set a default value in `_set_config()`. Currently if the value is not set the following error occurs:

```
Unable to get content from folder: /home/liipadmin/dropzones/DWH: 'delete_missing_datasets' / Traceback (most recent call last): File "/home/liipadmin/ogdzh/ckanext/ckanext-stadtzh-harvest/ckanext/stadtzhharvest/harvester.py", line 188, in gather_stage delete_ids = self._check_for_deleted_datasets(harvest_job) File "/home/liipadmin/ogdzh/ckanext/ckanext-stadtzh-harvest/ckanext/stadtzhharvest/harvester.py", line 1047, in _check_for_deleted_datasets if self.config['delete_missing_datasets']: KeyError: 'delete_missing_datasets'
--
```